From 522656bea15263ab90360eefe189dee655e37921 Mon Sep 17 00:00:00 2001
From: Peter Krempa <pkrempa@redhat.com>
Date: Mon, 12 Aug 2013 16:09:53 +0200
Subject: [PATCH 09/11] virsh-domain: Fix memleak in cmdUndefine with storage

When undefining a domain with storage when the volume isn't managed by
libvirt the name and path strings weren't freed properly.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=996050
(cherry picked from commit 5075248ac9e60dd73da3487f83be9aa188200688)
---
 tools/virsh-domain.c | 6 ++++++
 1 file changed, 6 insertions(+)

Index: libvirt-1.1.1/tools/virsh-domain.c
===================================================================
--- libvirt-1.1.1.orig/tools/virsh-domain.c	2014-03-25 14:15:18.882627367 -0500
+++ libvirt-1.1.1/tools/virsh-domain.c	2014-03-25 14:15:18.878627366 -0500
@@ -3048,6 +3048,8 @@ cmdUndefine(vshControl *ctl, const vshCm
 
         for (i = 0; i < nvolumes; i++) {
             ctxt->node = vol_nodes[i];
+            VIR_FREE(source);
+            VIR_FREE(target);
 
             /* get volume source and target paths */
             if (!(target = virXPathString("string(./target/@dev)", ctxt)))
@@ -3090,6 +3092,8 @@ cmdUndefine(vshControl *ctl, const vshCm
             }
             vlist[nvols].source = source;
             vlist[nvols].target = target;
+            source = NULL;
+            target = NULL;
             nvols++;
         }
 
@@ -3189,6 +3193,8 @@ out:
     }
 
 cleanup:
+    VIR_FREE(source);
+    VIR_FREE(target);
     for (i = 0; i < nvols; i++) {
         VIR_FREE(vlist[i].source);
         VIR_FREE(vlist[i].target);
